src/main/java/cc/polyfrost/oneconfig/config/Config.java

```diff
@@ -273,6 +287,131 @@ protected final void generateOptionList(Object instance, Class<?> targetClass, O
+                ConfigPageButton button;
+                if (pageInstance instanceof cc.polyfrost.oneconfig.gui.pages.Page) {
+                    button = new ConfigPageButton(field, instance, optionPage.name(), optionPage.description(), category, optionPage.subcategory(), (cc.polyfrost.oneconfig.gui.pages.Page) pageInstance);
+                } else {
+                    OptionPage newPage = new OptionPage(optionPage.name(), mod);
+                    generateOptionList(pageInstance, newPage, mod, migrate);
+                    button = new ConfigPageButton(field, instance, optionPage.name(), optionPage.description(), category, optionPage.subcategory(), newPage);
+                }
+                if (optionPage.location() == PageLocation.TOP) subcategory.topButtons.add(button);
+                else subcategory.bottomButtons.add(button);
+            } else if (field.isAnnotationPresent(HUD.class)) {
+                HUDUtils.addHudOptions(page, field, instance, this);
+            }
+            else if (field.isAnnotationPresent(SubCategory.class)){
+                SubCategory subcategoryAnnotation = field.getAnnotation(SubCategory.class);
+                String name = subcategoryAnnotation.name();
+                Object categoryInstance = ConfigUtils.getField(field, instance);
+
+                if(categoryInstance == null) continue;
+                if(categoryInstance instanceof cc.polyfrost.oneconfig.config.elements.OptionSubcategory) {
+                    page.categories.get(category).subcategories.add((cc.polyfrost.oneconfig.config.elements.OptionSubcategory) categoryInstance);
+                }
+                else{
+                    addSubCategory(categoryInstance,categoryInstance.getClass(),page,mod,category,name, migrate);
+                }
+            }
+        }
+        for (Method method : targetClass.getDeclaredMethods()) {
+            Button button = ConfigUtils.findAnnotation(method, Button.class);
+            String optionName = pagePath + method.getName();
+            if (button != null) {
+                BasicOption option = ConfigUtils.addOptionToPage(page, method, instance,category);
+                optionNames.put(optionName, option);
+            }
+        }
+        logger.trace("Finished generating option list for {} category (targetting={})", mod.name, targetClass.getName());
+    }
+
+    /**
+     * Adds a subcategory to the option list, for internal use only
+     *
+     * @param instance    instance of target class
+     * @param targetClass which class to lookup into
+     * @param page        page to add options too
+     * @param mod         data about the mod
+     * @param category    category name
+     * @param subcategory subcategory name
+     * @param migrate     whether the migrator should be run
+     */
+    void addSubCategory(Object instance, Class<?> targetClass, OptionPage page, Mod mod, String category, String subcategory, boolean migrate){
```

Comment:
this method is also only used once, see above comments about internal and typings.